### ISSUE TITLE
"repos" as an extract-like element 

### ISSUE THREAD

**Issue by hlesaint** [openmindanesth/openmind#89]
Currently, OpenMind focuses on key findings with a link to the original data. It would be great to also implement an extract-like element to simply refer to repositories of briefly described raw data, e.g. fMRI resting state, 20 male mice, 9.4T small animal scanner, heart rate measured, etc. Figure is a representative image to show data quality and give a visual clue.

This type of extract is for the big shots, who have tons of data but no time to look into it. Having the choice to contribute with such "repo cards" will motivate many to share their data who consider the restriction for contributing with findings a barrier. These "repo cards" enable others to mine the repositories that are now available and create extracts on specific effects. 

Both, making data available and sharing insights on particular effects will take us forward. We should make OpenMind a hub for both.

**Comment by hlesaint** [comment]
Ideally we should make it possible to link these "repo cards" to the repo section of an extract (if it uses data from that repo) and maybe totally replacing the old way of typing repos directly into the extract.

**Comment by tgetgood** [comment]
This is a good idea. 

How would we incorporate data into search? My intuition is that we would want to have a completely orthogonal search. Something like search by: articles, labnotes, findings (articles or labnotes), or data set. Mixing datasets into the rest might prove confusing. 

For creation I could move the article/labnote selector to the very top and add a third option, or I could allow you to create a data set inline as you create an extract. Maybe both are useful. 

We would need a robust search over datasets to make sure people don't add the same dataset more than once. 

I think that if we're going to reify repos like this we should be uniform about it. Make every repo an independent entity (right now, figures with captions are such an independent entity, though it's invisible) and link them back to extracts.

What would be the types of repos? How much info should we collect?

**Comment by hlesaint** [comment]
I fully agree on a search of either: articles, labnotes, findings (articles or labnotes), or repos. We further should have an option to search for findings that are linked to one particular repo card.

I also think both are useful: a separate 3rd option on top, and to create a repo inline as you create an extract.

Looking for having the same repo twice should be straightforward by comparing the link to datasets, code, toolboxes as they are usually uploaded at external repositories from the person/group that created it in the first place. 

I think 2 types of repo cards would suffice: 
- datasets 
- code & toolboxes

<br>

A _**dataset** repo card_ should entail 

(1) links to a set of interdependent data that is e.g.,
  - the same experiment conducted across n subjects
  - simultaneously acquired data like fMRI, heart and breathing rate
  - arguably: successively acquired data of the same experiment, but with different modalities, e.g., EEG and fMRI
  - arguably: the toolboxes /code that brought the data in its accessible form

(2) a short description of 
- the actual content in on data in the repository (fMRI scans, EEG, heart rate, etc.) 
- the experimental settings (9.4T Bruker biospin, 32-channels, pulse oximetry, etc.)
- investigator(s), year, institute, department (as for lab notes) 
- anything which is important as formless and free text
- If the data is uploaded in [bids format](https://github.com/bids-standard/bids-starter-kit/wiki/The-BIDS-folder-hierarchy) the actual content and the experimental settings can be at least partially parsed and automatically filled in based on the data structure and json files (see https://openneuro.org/).

(3) an example figure that shows 
- data content
- data quality 

<br>

A _**code & toolbox** repo card_ should entail 

(1) links to code and toolboxes that are used to process/analyze uploaded data to create specific findings of an extract
(2) a short description of these scripts and toolboxes
(3) an optional example figure

<br>

Both types of repo cards should entail similar features as usual extracts: comments, history, related, repos, tags, author (year)
